WordPress/src/main/java/org/wordpress/android/ui/engagement/ListScenarioUtils.kt

```diff
@@ -0,0 +1,93 @@
+package org.wordpress.android.ui.engagement
+
+import android.app.Activity
+import android.text.Spannable
+import dagger.Reusable
+import org.json.JSONArray
+import org.json.JSONException
+import org.wordpress.android.fluxc.tools.FormattableContent
+import org.wordpress.android.models.Note
+import org.wordpress.android.ui.engagement.ListScenarioType.LOAD_COMMENT_LIKES
+import org.wordpress.android.ui.engagement.ListScenarioType.LOAD_POST_LIKES
+import org.wordpress.android.ui.engagement.AuthorName.AuthorNameCharSequence
+import org.wordpress.android.ui.notifications.blocks.HeaderNoteBlock
+import org.wordpress.android.ui.notifications.blocks.NoteBlockClickableSpan
+import org.wordpress.android.ui.notifications.utils.NotificationsUtilsWrapper
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T.NOTIFS
+import org.wordpress.android.util.getMediaUrlOrEmpty
+import org.wordpress.android.util.getRangeIdOrZero
+import org.wordpress.android.util.getRangeSiteIdOrZero
+import org.wordpress.android.util.getRangeUrlOrEmpty
+import org.wordpress.android.util.getTextOrEmpty
+import org.wordpress.android.util.image.ImageManager
+import org.wordpress.android.util.image.ImageType.AVATAR_WITH_BACKGROUND
+import java.util.ArrayList
+import javax.inject.Inject
+
+@Reusable
+class ListScenarioUtils @Inject constructor(
+    val imageManager: ImageManager,
+    val notificationsUtilsWrapper: NotificationsUtilsWrapper
+) {
+    fun mapLikeNoteToListScenario(note: Note, activity: Activity): ListScenario {
+        if (!note.isLikeType) throw IllegalArgumentException(
+                "mapLikeNoteToListScenario > unexpected note type ${note.type}"
+        )
+
+        val imageType = AVATAR_WITH_BACKGROUND
+        val headerNoteBlock = HeaderNoteBlock(
+                null,
+                transformToFormattableContentList(note.header),
+                imageType,
+                null,
+                null,
+                imageManager,
+                notificationsUtilsWrapper
+        )
+        headerNoteBlock.setIsComment(note.isCommentType)
+
+        val spannable: Spannable = notificationsUtilsWrapper.getSpannableContentForRanges(headerNoteBlock.getHeader(0))
+        val spans = spannable.getSpans(
+                0, spannable.length,
+                NoteBlockClickableSpan::class.java
+        )
+        for (span in spans) {
+            span.enableColors(activity)
```

Comment:
I think we can just use a Context for this one.

Reply:
👍 ! Just noting I initially tried to use the injected ContextProvider (that is an Application Context) but AFAIU I think we need a full activity/view context since we use `getColorFromAttribute` calls inside the `span.enableColors`